back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -217,11 +216,15 @@ def update
 
     user_can_moderate_project = UserRoleService.new.can_moderate_project?(input.project, current_user)
     update_params = idea_params(input.custom_form, user_can_moderate_project).to_h
+    phase_ids = update_params.delete(:phase_ids) if update_params[:phase_ids]
     update_params[:custom_field_values] = params_service.updated_custom_field_values(input.custom_field_values, update_params[:custom_field_values])
     CustomFieldService.new.compact_custom_field_values! update_params[:custom_field_values]
+    input.set_manual_votes(update_params[:manual_votes_amount], current_user) if update_params[:manual_votes_amount]
 
     ActiveRecord::Base.transaction do # Assigning relationships cause database changes
       input.assign_attributes update_params
+      sidefx.before_update(input, current_user)
+      input.phase_ids = phase_ids if phase_ids
```

Comment:
We don't want to set the phase_ids before before_update, because then the service can't figure out what the previous values were. But we do want to assign them in the transaction, because this causes database changes (as mentioned in the comment).